javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/InternalLogger.java

```diff
@@ -28,11 +28,11 @@ static InternalLogger getLogger(String name) {
     return loggerFactory.get().create(name);
   }
 
-  protected abstract boolean isLoggable(Level level);
+  public abstract boolean isLoggable(Level level);
 
-  protected abstract void log(Level level, String message, @Nullable Throwable error);
+  public abstract void log(Level level, String message, @Nullable Throwable error);
 
-  protected abstract String name();
+  public abstract String name();
```

Comment:
My preference is to never use inheritance for this kind of thing. Would prefer to make `InternalLogger` final and concrete, and turn these 3 methods into an interface called maybe `InternalLoggingStrategy` and then the concrete impl gets passed an instance of that strategy to delegate to.

Reply:
`InternalLogger` only contains abstract methods though. AFAIR I only made it an abstract class to avoid putting the `InternalLogger.Factory` global variable in some other class; I guess I could do that and convert it to an interface. WDYT?

Reply:
Yeah, I'd like to see that.